src/RefractAPI.cc

```diff
@@ -19,551 +21,537 @@
     }
 
-    refract::IElement* ParameterToRefract(const NodeInfo<snowcrash::Parameter>& parameter, ConversionContext& context)
+    // Parameter use
     {
-        refract::MemberElement* element = new refract::MemberElement;
-        refract::IElement* value = ExtractParameter(parameter, context);
-        element->set(PrimitiveToRefract(MAKE_NODE_INFO(parameter, name)), value);
-
-        // Description
-        if (!parameter.node->description.empty()) {
-            element->meta[SerializeKey::Description] = PrimitiveToRefract(MAKE_NODE_INFO(parameter, description));
-        }
-
-        if (!parameter.node->type.empty()) {
-            element->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_NODE_INFO(parameter, type));
-        }
 
-        // Parameter use
         std::string use = SerializeKey::Required;
 
         if (parameter.node->use == snowcrash::OptionalParameterUse) {
             use = SerializeKey::Optional;
         }
 
-        refract::ArrayElement* typeAttributes = new refract::ArrayElement;
+        element->attributes().set(SerializeKey::TypeAttributes, CreateArrayElement(make_primitive(use)));
+    }
 
-        typeAttributes->push_back(refract::IElement::Create(use));
-        element->attributes[SerializeKey::TypeAttributes] = typeAttributes;
+    return std::move(element);
+}
 
-        return element;
-    }
+std::unique_ptr<IElement> ParametersToRefract(
+    const NodeInfo<snowcrash::Parameters>& parameters, ConversionContext& context)
+{
+    return CollectionToRefract<ObjectElement>(parameters, context, ParameterToRefract, SerializeKey::HrefVariables);
+}
 
-    refract::IElement* ParametersToRefract(
-        const NodeInfo<snowcrash::Parameters>& parameters, ConversionContext& context)
-    {
-        return CollectionToRefract<refract::ObjectElement>(
-            parameters, context, ParameterToRefract, SerializeKey::HrefVariables);
-    }
+std::unique_ptr<IElement> HeaderToRefract(const NodeInfo<snowcrash::Header>& header, ConversionContext& context)
+{
+    auto element = make_element<MemberElement>(make_primitive(header.node->first), make_primitive(header.node->second));
 
-    refract::IElement* HeaderToRefract(const NodeInfo<snowcrash::Header>& header, ConversionContext& context)
-    {
-        refract::MemberElement* element = new refract::MemberElement;
+    AttachSourceMap(*element, header);
+
+    return std::move(element);
+}
+
+std::unique_ptr<IElement> AssetToRefract(
+    const NodeInfo<snowcrash::Asset>& asset, const std::string& contentType, const std::string& metaClass)
+{
+    if (asset.node->empty()) {
+        return nullptr;
+    }
 
-        element->set(refract::IElement::Create(header.node->first), refract::IElement::Create(header.node->second));
+    auto element = PrimitiveToRefract(asset);
 
-        AttachSourceMap(element, header);
+    element->element(SerializeKey::Asset);
+    element->meta().set(SerializeKey::Classes, CreateArrayElement(make_primitive(metaClass)));
```

Comment:
In general, I understand, it will create "primitive element", but It is little bit confusing to have `make_primitive()` and `PrimitiveToRefract()`
I propose something like `from_primitive()`

And there is still problem with naming Convention (see `is_primitive()`)